File: frontend/frontend/src/components/UserService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/users';

const addBookToUser = (userId, book) => {
    return axios.post(`${API_URL}/${userId}/books`, book);
};

const startReadingBook = (userId, bookId) => {
    return axios.post(`${API_URL}/${userId}/books/${bookId}/start`);
};

const finishReadingBook = (userId, bookId) => {
    return axios.post(`${API_URL}/${userId}/books/${bookId}/finish`);
};

const getBookStatus = (userId, bookId) => {
    return axios.get(`${API_URL}/${userId}/books/${bookId}/status`);
};

const getUserBooks = (userId) => {
    return axios.get(`${API_URL}/${userId}/books`);
};

export default {
    addBookToUser,
    startReadingBook,
    finishReadingBook,
    getBookStatus,
    getUserBooks
};

File: web/src/main/java/com/example/web/services/UserService.java
package com.example.web.services;

import java.util.HashMap;
import java.util.Map;

import com.example.web.models.Book;
import com.example.web.models.BookStatus;
import com.example.web.models.User;

public class UserService {
    private Map<String, User> users;

    public UserService() {
        users = new HashMap<>();
    }

    public User getUser(String userId) {
        return users.get(userId);
    }

    public void addUser(User user) {
        if (!users.containsKey(user.getId())) {
            users.put(user.getId(), user);
        }
    }

    public void addBookToUser(String userId, Book book) {
        User user = getUser(userId);
        if (user != null) {
            user.addBook(book);
        }
    }

    public void startReadingBook(String userId, String bookId) {
        User user = getUser(userId);
        if (user != null) {
            user.startReadingBook(bookId);
        }
    }

    public void finishReadingBook(String userId, String bookId) {
        User user = getUser(userId);
        if (user != null) {
            user.finishReadingBook(bookId);
        }
    }

    public BookStatus getBookStatus(String userId, String bookId) {
        User user = getUser(userId);
        if (user != null) {
            return user.getBookStatus(bookId);
        }
        return BookStatus.NONE;
    }
}
